velox/exec/OrderBy.cpp

```diff
@@ -91,12 +91,7 @@ OrderBy::OrderBy(
   }
 #endif
 
-  outputBatchSize_ = std::max<uint32_t>(
-      operatorCtx_->execCtx()
-          ->queryCtx()
-          ->queryConfig()
-          .preferredOutputBatchSize(),
-      data_->estimatedNumRowsPerBatch(kBatchSizeInBytes));
+  outputBatchSize_ = outputBatchRows(data_->fixedRowSize());
```

Comment:
Why data_->fixedRowSize()?

Reply:
Also tried the alternative way to pass RowContainer::estimateRowSize() that I added. The impl of RowContainer::estimateRowSize() is based on allocatedBytes.
https://github.com/facebookincubator/velox/blob/02844168ae12d1a3c6372f2605f1e2e767d8b933/velox/exec/RowContainer.cpp#L564-L575

Here we're at the initializer of OrderBy, and the allocatedBytes is 0 and so estimateRowSize() returns 0. This makes outputBatchRows(RowContainer::estimateRowSize()) returns 1024, kPreferredOutputBatchRows.

For the impl of  RowContainer::estimateRowSize(), I'm wondering if I should return fixedRowSize_, when numRows_ is 0.

Reply:
Per offline discussion, we will change the logic to `outputBatchSize_ = outputBatchRows();`, as the original logic doesn't take into account variable-length rows and perhaps also needs to be improved. Will file an Issue and revisit it with @xiaoxmeng .